bin/improver-spot-extract

```diff
@@ -122,6 +149,40 @@ def main():
         grid_metadata_identifier=args.grid_metadata_identifier)
     result = plugin.process(neighbour_cube, diagnostic_cube)
 
+    # If a probability or percentile diagnostic cube is provided, extract
+    # the given percentile if available. This is done after the spot-extraction
+    # to minimise processing time; usually there are far fewer spot sites than
+    # grid points.
+    if args.extract_percentile:
+        try:
+            perc_coordinate = find_percentile_coordinate(result)
+        except CoordinateNotFoundError:
+            if 'probability_of_' in result.name():
+                result = GeneratePercentilesFromProbabilities(
+                    ecc_bounds_warning=args.ecc_bounds_warning).process(
+                        result, percentiles=args.extract_percentile)
+                result = iris.util.squeeze(result)
+            else:
+                msg = ('Diagnostic cube is not a known probabilistic type. '
```

Comment:
What if we have realization input, do we want to extract percentiles from this too?

Reply:
Added. Highlights some issues that might want attention regarding the percentile coordinate name and units when produced using PercentileConverter.